ui/src/errors.ts

```diff
@@ -0,0 +1,20 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { localize } from "./localize";
+
+export class UserCancelledError extends Error { }
+
+export class NotImplementedError extends Error {
+    constructor(methodName: string, obj: object) {
+        super(localize('azFunc.notImplementedError', '"{0}" is not implemented on "{1}".', methodName, obj.constructor.name));
+    }
+}
+
+export class ArgumentError extends Error {
+    constructor(obj: object) {
```

Comment:
What if obj is null/undefined?

Reply:
This repo has 'strictNullChecks' on and this is an 'internal' class (aka it's not in index.d.ts at the root of the repo), so that should never happen